torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,532 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""IDTReeS dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from rasterio.enums import Resampling
+from torch import Tensor
+from torchvision.utils import draw_bounding_boxes
+
+from .geo import VisionDataset
+from .utils import download_url, extract_archive
+
+
+class IDTReeS(VisionDataset):
+    """IDTReeS dataset.
+
+    The `IDTReeS <https://idtrees.org/competition/>`_
+    dataset is a dataset for tree crown detection.
```

Comment:
Can you add more details like image source, image size, number of images, etc.? Basically, I would like to have everything that we put in Table 1 of our paper listed for every dataset, although plenty of existing datasets are missing this info.